clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerProtocolTest.java

```diff
@@ -137,6 +161,19 @@ public void deserializeFutureSubscriptionVersion() {
         assertEquals(groupInstanceId, subscription.groupInstanceId());
     }
 
+    @Test
+    public void serializeDeserializeAssignmentAllVersions() {
+        List<TopicPartition> partitions = Arrays.asList(tp1, tp2);
+        Assignment assignment = new Assignment(partitions, ByteBuffer.wrap("hello".getBytes()));
+
+        for (short version = ConsumerProtocolAssignment.LOWEST_SUPPORTED_VERSION; version <= ConsumerProtocolAssignment.HIGHEST_SUPPORTED_VERSION; version++) {
+            ByteBuffer buffer = ConsumerProtocol.serializeAssignment(assignment, version);
+            Assignment parsedAssignment = ConsumerProtocol.deserializeAssignment(buffer);
+            assertEquals(toSet(partitions), toSet(parsedAssignment.partitions()));
+            assertEquals(parsedAssignment.userData(), parsedAssignment.userData());
```

Comment:
Does it compare to incorrect object (maybe ```assignment.userData()```)?

Reply:
Good catch!